src/core/Serialize.ml

```diff
@@ -474,14 +473,19 @@ struct
   and json_of_env {tpenv; conenv} : J.value =
     `O [("tpenv", json_of_bwd json_of_tp tpenv); ("conenv", json_of_bwd json_of_con conenv)]
 
+  and json_of_cof_var : D.cof_var -> J.value  = 
+    function
+    | CofVar.Local lvl -> labeled "local" [json_of_int lvl]
+    | CofVar.Axiom sym -> labeled "axiom" [Global.serialize sym]
+
   and json_of_cof (cof : D.cof) : J.value =
-    Cof.json_of_cof json_of_dim json_of_int cof
+    Cof.json_of_cof json_of_dim json_of_cof_var cof
 
   and json_of_dim : D.dim -> J.value =
     function
     | Dim0 -> `String "dim0"
     | Dim1 -> `String "dim1"
-    | DimVar n -> labeled "dim_var" [json_of_int n]
+    | DimVar n -> labeled "dim_var" [json_of_cof_var n]
```

Comment:
Is there a mismatch here? `dim_var` vs `cof_var`? Maybe it is just identifier naming.

Reply:
This is correct, albeit a bit weird. This would serialize to `["dim_var", ["local", 1]]` which follows the AST structure.